src/UriFactory/GuzzleFactory.php

```diff
@@ -0,0 +1,22 @@
+<?php
+
+namespace Http\Discovery\UriFactory;
+
+use GuzzleHttp\Psr7\Uri;
+use Http\Message\UriFactory;
+
+/**
+ * Creates a guzzlehttp/psr7 URI object
+ *
+ * @author David de Boer <david@ddeboer.nl>
+ */
+class GuzzleFactory implements UriFactory
+{
+    /**
+     * {@inheritdoc}
+     */
+    public function createUri($uri)
+    {
```

Comment:
Yup. The reason why UriInterface is accepted in the factory interface is that you can accept both string and uri implementation in your library and you can leave the check to the factory.